crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -227,99 +236,103 @@ pub trait LoadPendingBlock:
         let mut sum_blob_gas_used = 0;
         let block_gas_limit: u64 = block_env.gas_limit;
 
-        let mut best_txs =
-            self.pool().best_transactions_with_attributes(BestTransactionsAttributes::new(
-                block_env.basefee,
-                block_env.blob_gasprice().map(|gasprice| gasprice as u64),
-            ));
-
-        while let Some(pool_tx) = best_txs.next() {
-            // ensure we still have capacity for this transaction
-            if cumulative_gas_used + pool_tx.gas_limit() > block_gas_limit {
-                // we can't fit this transaction into the block, so we need to mark it as invalid
-                // which also removes all dependent transaction from the iterator before we can
-                // continue
-                best_txs.mark_invalid(
-                    &pool_tx,
-                    InvalidPoolTransactionError::ExceedsGasLimit(
-                        pool_tx.gas_limit(),
-                        block_gas_limit,
-                    ),
-                );
-                continue
-            }
-
-            if pool_tx.origin.is_private() {
-                // we don't want to leak any state changes made by private transactions, so we mark
-                // them as invalid here which removes all dependent transactions from the iterator
-                // before we can continue
-                best_txs.mark_invalid(
-                    &pool_tx,
-                    InvalidPoolTransactionError::Consensus(
-                        InvalidTransactionError::TxTypeNotSupported,
-                    ),
-                );
-                continue
-            }
+        // Only include transactions if not configured as Empty
+        if self.pending_block_config() != PendingBlockConfig::Empty {
```

Comment:
nit: can we introduce `fn is_|empty|none` functions so we can replace these == with function calls